Add support for anyof_ref

gmlewis opened the pull request:
Fixes: #71.

samlown reviewed (approved):
Looks fine. I suspect this might have conflicts with #69 (`ipAddress` could be `ip_address` to be more consistent with the other examples)

gmlewis commented:
> Looks fine. I suspect this might have conflicts with #69 (`ipAddress` could be `ip_address` to be more consistent with the other examples)

These two PRs should actually merge nicely with each other.  I first implemented the feature in #69 and later added this feature, but both work together in my repo.  I'll help with the merging when it comes time and resolve any conflicts.

Sorry for the delay.  I addressed the review feedback. PTAL.

samlown commented on README.md:
```diff
@@ -37,6 +37,8 @@ type TestUser struct {
   BirthDate     time.Time              `json:"birth_date,omitempty" jsonschema:"oneof_required=date"`
   YearOfBirth   string                 `json:"year_of_birth,omitempty" jsonschema:"oneof_required=year"`
   Metadata      interface{}            `json:"metadata,omitempty" jsonschema:"oneof_type=string;array"`
+  IPAddress2    interface{}            `json:"ip_address2,omitempty" jsonschema:"anyof_ref=#/$defs/ipv4;#/$defs/ipv6"`
```
I don't think these are important enough to stick in the README. I think the best way forward here is with some custom tests inside the refect_test.go file.
